packages/protocol/contracts/L2/TaikoL2.sol

```diff
@@ -124,6 +125,11 @@ contract TaikoL2 is CrossChainOwned, ICrossChainSync {
             parentId = block.number - 1;
         }
 
+        // Verify parentBlockHader match the block hash
+        if (parentBlockHeader.hashBlockHeader() != blockhash(parentId)) {
```

Comment:
We are on L2 and it seems with this you are comparing L2, but should not we need L1 ? 🤔